### PR TITLE
Add type option to instance creation.

### PR DESCRIPTION
**Pull request by kolodny**
- [x] Tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -388,6 +388,21 @@ describe('Bigtable', function() {
       bigtable.createInstance(INSTANCE_NAME, options, assert.ifError);
     });
 
+    it('should respect the type option', function(done) {
+      var options = {type: 'development'};
+      sinon.spy(FakeInstance, 'getTypeType_');
+
+      bigtable.request = function(config) {
+        assert.deepEqual(config.reqOpts.instance.type, 2);
```
Comment (stephenplusplus) [inline test/index.js]: This line tests functionality of `Instance.getTypeType_`. We need another test block for the new `Instance.getTypeType_` function to make sure that works as intended.

File: test/index.js [hunk 2]
```diff
@@ -388,6 +388,21 @@ describe('Bigtable', function() {
       bigtable.createInstance(INSTANCE_NAME, options, assert.ifError);
     });
 
+    it('should respect the type option', function(done) {
+      var options = {type: 'development'};
+      sinon.spy(FakeInstance, 'getTypeType_');
```
Comment (stephenplusplus) [inline test/index.js]: We need to simulate what this method returns, so we can verify that's what ends up being sent to `request`.

For example, this pattern:

```js
var options = {type: 'development'}
var fakeTypeType = 99

FakeInstance.getTypeType_ = function(options) {
  assert.strictEqual(type, options.type)
  return fakeTypeType
}

bigtable.request = function(config) {
  assert.strictEqual(config.reqOpts.instance.type, fakeTypeType)
}

bigtable.createInstance(INSTANCE_NAME, options, assert.ifError)
```

File: test/index.js (hunk 1 above)
Comment (kolodny) [reply test/index.js]: Added a test in https://github.com/googleapis/nodejs-bigtable/pull/110/files#diff-c5fd52f715b6d5d9a356e4b76caf7511

File: test/index.js (hunk 2 above)
Comment (kolodny) [reply test/index.js]: Done

File: test/instance.js [hunk 3]
```diff
@@ -127,6 +127,28 @@ describe('Bigtable/Instance', function() {
...
+      production: 1,
+      development: 2,
+    };
+
+    it('should default to unspecified', function() {
+      assert.strictEqual(Instance.getTypeType_(), types.unspecified);
+    });
+
+    it('should lowercase a type', function() {
+      assert.strictEqual(Instance.getTypeType_('production'), types.production);
```
Comment (stephenplusplus) [inline test/instance.js]: Did you mean `'PRODUCTION'`?

File: test/instance.js [hunk 4]
```diff
@@ -127,6 +127,28 @@ describe('Bigtable/Instance', function() {
...
 
+  describe('getTypeType_', function() {
+    var types = {
+      unspecified: 0,
+      production: 1,
+      development: 2,
+    };
+
+    it('should default to unspecified', function() {
+      assert.strictEqual(Instance.getTypeType_(), types.unspecified);
```
Comment (stephenplusplus) [inline test/instance.js]: You could add an extra test here like:

```js
assert.strictEqual(Instance.getTypeType_('not-real-type'), types.unspecified)
```

File: test/instance.js (hunk 3 above)
Comment (kolodny) [reply test/instance.js]: Yes, good catch

File: test/instance.js (hunk 4 above)
Comment (kolodny) [reply test/instance.js]: Sounds good!

**Pull request merged**